test/represent-test.coffee

```diff
@@ -0,0 +1,159 @@
+                              "typeSpecification": {
+                                "name": "object",
+                                "nestedTypes": []
+                              },
+                              "attributes": []
+                            }
+                          },
+                          "sections": [
+                            {
+                              "class": "memberType",
+                              "content": [
+                                {
+                                  "content": {
+                                    "name": {
+                                      "literal": "id"
+                                    },
+                                    "description": "",
+                                    "valueDefinition": {
+                                      "values": [
+                                        {
+                                          "literal": "pavan",
+                                          "variable": false
+                                        }
+                                      ],
+                                      "typeDefinition": {
+                                        "typeSpecification": {
+                                          "name": null,
+                                          "nestedTypes": []
+                                        },
+                                        "attributes": []
+                                      }
+                                    },
+                                    "sections": []
+                                  },
+                                  "class": "property"
+                                }
+                              ]
+                            }
+                          ]
+                        },
+                        "class": "value"
+                      },
+                      {
+                        "content": {
+                          "description": "",
+                          "valueDefinition": {
+                            "values": [
+                              {
+                                "literal": "pkumar",
+                                "variable": false
+                              }
+                            ],
+                            "typeDefinition": {
+                              "typeSpecification": {
+                                "name": null,
+                                "nestedTypes": []
+                              },
+                              "attributes": []
+                            }
+                          },
+                          "sections": []
+                        },
+                        "class": "value"
+                      }
+                    ]
+                  }
+                ]
+              },
+              "class": "property"
+            }
+          ]
+        }
+      ]
+    }
```

Comment:
you can provide the same test with using the current testsuite:

- put AST into `test/formats/samples-ast` as a new sample under name of your selection
- put rendered JSON into `test/formats/samples-json` under the same name
- add the test case to `formats-test.coffee`

Also please test all the situations you fixed - `oneOf` for JSON Schema, `oneOf` for JSON, etc. How current testsuite basically works (this could be in the README, shame on me :-/ ):

- you write `.mson` samples for your test cases
- you run [`scripts/generate-samples-ast.coffee`](https://github.com/apiaryio/boutique.js/blob/master/scripts/generate-samples-ast.coffee) from command line - it should generate AST samples for you
- you check the AST samples
- you write down expected JSON and JSON Schema samples
- you add these new test cases to the `formats-test.coffee` file
- you run tests

The problem is that the script is still for protagonist-experimental instead of drafter, but that can be easily fixed I guess. Feel free to ask in case you're in troubles :)